Add note about January 2021 fixes to jar signer

zathras opened the pull request:
This repos comes up in searches, and is fairly widely followed.  Could you add this note, please?  If you like, you're more than welcome to incorporate the JAR signer fix.  It's pretty simple - it's in the history, and I'd be happy to discuss if I could figure out your contact info :-)

zathras committed:
Add note about January 2021 fixes to jar signer

oliverlietz commented:
@zathras, Which JDK are you using? With both Oracle JDK and OpenJDK I get `package sun.security.tools.keytool does not exist`. The package is not part of `tools.jar`.
```
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.321-b07, mixed mode)
```
```
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-bre_2021_10_20_23_15-b00)
OpenJDK 64-Bit Server VM (build 25.312-b00, mixed mode)
```

zathras commented:
That's odd.  I double-checked on Linux, and (after I fixed an ancient bug in the clean target due to the case-sensitive filesystem), it worked for me on Linux, as well as MacOS.  I added the version details to https://github.com/zathras/java.net/blob/master/hdcookbook/README.md

Are you using ant to build?  I had to add a javac command-line argument to allow access to more of the sun.* stuff.  With that said, there might be other options the ant build passes in, so this might or might not be the problem.

Ah, there it is:  -XDignore.symbol.file in the build.xml for the JAR signer tool.  I added a note about this to the README.md.

oliverlietz commented:
No Ant anymore, I changed the build to use Maven 10 years ago. The additional compiler argument (`-XDignore.symbol.file`) is key and I had to add an additional system dependency on `rt.jar` (contains package `sun.security.tools.keytool`).

zathras commented:
Makes sense.  I had to figure out where to get ant for MacOS to get the original to work.

oliverlietz commented:
[Ant](https://ant.apache.org) is available via [Homebrew](https://brew.sh): https://formulae.brew.sh/formula/ant#default

zathras commented:
Yes, that's where I found it too.

But anyway, could you add links to the archive of the original repo and the archive of hdcookbook.com to the README, please?  It's good to leave breadcrumbs, in case someone wants to reconstruct this stuff some day.  That's basically why I fixed it for 1.8.

oliverlietz commented:
@zathra, Sure. Now that schemagen and security are fixed and java.net *and* hdcookbook.com are dead I will do some clean up and start to modernize Authoring and Disc Creation Tools and [site](https://oliverlietz.github.io/bd-j/).
